notification-portlet-webapp/src/main/java/org/jasig/portlet/notice/service/jpa/action/CompleteOnRedirectAction.java

```diff
@@ -60,27 +61,11 @@ public StateChangeAction(String label) {
     @Override
     public void invoke(final ActionRequest req, final ActionResponse res) throws IOException {
         JpaNotificationService jpaService = (JpaNotificationService) SpringContext.getApplicationContext().getBean("jpaNotificationService");
-        CacheNotificationService cacheService = (CacheNotificationService) SpringContext.getApplicationContext().getBean("cacheNotificationService");
-
-        final PortletPreferences prefs = req.getPreferences();
-        final String clickedState = prefs.getValue("StateChangeNotificationServiceDecorator.applyState", "COMPLETED");
-        NotificationState notificationState = NotificationState.valueOf(clickedState);
-
-        boolean stateFound = false;
 
         final NotificationEntry entry = getTarget();
         Map<NotificationState, Date> stateMap = entry.getStates();
-        if (stateMap != null && stateMap.size() > 0) {
-            for (NotificationState state : stateMap.keySet()) {
-                if (state == notificationState) {
-                    stateFound = true;
-                }
-            }
-        }
-
-        if (!stateFound) {
-            jpaService.addEntryState(req, entry.getId(), notificationState);
-            cacheService.clearCacheForUser(req);
+        if (stateMap != null && !stateMap.containsKey(NotificationState.COMPLETED)) {
```

Comment:
Nitpick:  Lost the ability to configure the state to add.  Having said that, not sure any other state wouldn't require additional processing.